libs/ui-components/src/components/common/PermissionsContext.tsx

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * Helper function to check if a permission (resource + verb) is allowed
+ * Handles wildcards: "*" in resource means all resources, "*" in operations means all operations
+ */
+const isPermissionAllowed = (userPermissions: Permission[], permissionCheck: PermissionCheck): boolean => {
+  for (const permission of userPermissions) {
+    // Check if resource matches (exact match or wildcard)
+    const resourceMatches = permission.resource === '*' || permission.resource === permissionCheck.kind;
+    if (resourceMatches) {
+      // Check if operation/verb matches (exact match or wildcard)
+      const verbMatches = permission.operations.includes('*') || permission.operations.includes(permissionCheck.verb);
+      if (verbMatches) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+export type PermissionsContextType = {
+  loading: boolean;
+  error: string | undefined;
+  checkPermissions: (permissions: PermissionCheck[]) => boolean[];
+};
+
+const PermissionsContext = React.createContext<PermissionsContextType | undefined>(undefined);
+
+export const PermissionsContextProvider = ({ children }: React.PropsWithChildren) => {
+  const [userPermissions, setUserPermissions] = React.useState<Permission[]>([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState<string | undefined>();
+  const { currentOrganization } = useOrganizationGuardContext();
+
+  const { get } = useFetch();
+
+  React.useEffect(() => {
+    const abortController = new AbortController();
+
+    const loadPermissions = async () => {
+      // Only fetch permissions when an organization has been selected
+      if (!currentOrganization) {
+        setLoading(false);
+        setUserPermissions([]);
+        return;
+      }
+
+      setLoading(true);
+      setError(undefined);
+      try {
+        const response = await get<PermissionList>('auth/listpermissions', abortController.signal);
+        if (!abortController.signal.aborted) {
+          setUserPermissions(response.permissions);
+        }
+      } catch (err) {
+        // Ignore abort errors - they're expected when organization changes
+        if (!abortController.signal.aborted) {
+          setError(err instanceof Error ? err.message : 'Failed to fetch permissions');
+          setUserPermissions([]);
+        }
+      } finally {
+        if (!abortController.signal.aborted) {
+          setLoading(false);
+        }
+      }
+    };
+
+    void loadPermissions();
+
+    // Cleanup: abort the request if the organization changes or component unmounts
+    return () => {
+      abortController.abort();
+    };
+  }, [get, currentOrganization]);
+
+  const checkPermissions = React.useCallback(
+    (permissionChecks: PermissionCheck[]): boolean[] => {
+      if (permissionChecks.length === 0) {
+        return [];
+      }
+
+      return permissionChecks.map((permissionCheck) => isPermissionAllowed(userPermissions, permissionCheck));
```

Comment:
the length check is not needed
```suggestion
      return permissionChecks.map((permissionCheck) => isPermissionAllowed(userPermissions, permissionCheck));
```

Reply:
Fixed.